src/redux/features/rockets/rocketsSlice.js

```diff
@@ -13,6 +13,26 @@ export const fetchRockets = createAsyncThunk('rocket/fetchRocket', () => axios.g
 const rocketSlice = createSlice({
   name: 'rocket',
   initialState,
+  reducers: {
+    booked: (state, action) => {
+      const id = action.payload;
+      state.rockets = state.rockets.map((rocket) => {
+        if (rocket.id === id) {
+          return { ...rocket, reserved: true };
+        }
+        return rocket;
+      });
+    },
+    cancelReservation: (state, action) => {
+      const id = action.payload;
+      state.rockets = state.rockets.map((rocket) => {
+        if (rocket.id === id) {
+          return { ...rocket, reserved: false };
+        }
+        return rocket;
+      });
+    },
+  },
   extraReducers: (builder) => {
     builder.addCase(fetchRockets.pending, (state) => {
```

Comment:
Could you please add reject builder on extra reducers that shows if the API isn't working properly